crates/sui-types/src/gas.rs

```diff
@@ -452,20 +453,20 @@ pub fn check_gas_balance(
         });
     }
 
-    let max_gas_budget = cost_table.max_gas_budget;
-    let min_gas_budget = cost_table.min_gas_budget_external();
+    let max_gas_budget = cost_table.max_gas_budget as u128 * gas_price as u128;
+    let min_gas_budget = cost_table.min_gas_budget_external() as u128 * gas_price as u128;
```

Comment:
max and min gas budget are unchanged for now. I think we will review that in time but it should not be a problem.
I think gas price used to be 1 almost everywhere so it did not make a material difference but we will review in time